File: Android/TUIKit/TUICallKit/tuicallkit-kt/src/main/java/com/tencent/qcloud/tuikit/tuicallkit/state/ViewState.kt
package com.tencent.qcloud.tuikit.tuicallkit.state

import com.trtc.tuikit.common.livedata.LiveData

class ViewState {
    var reverse1v1CallRenderView = false
    var showLargeViewUserId: LiveData<String> = LiveData("")
    var isVirtualBackgroundOpened: LiveData<Boolean> = LiveData(false)
    val isScreenCleaned: LiveData<Boolean> = LiveData(false)
    val router: LiveData<ViewRouter> = LiveData(ViewRouter.None)
    var enterPipMode: LiveData<Boolean> = LiveData(false)

    fun reset() {
        reverse1v1CallRenderView = false
        showLargeViewUserId.set("")
        isVirtualBackgroundOpened.set(false)
        isScreenCleaned.set(false)
        router.set(ViewRouter.None)
        enterPipMode.set(false)
    }

    enum class ViewRouter {
        None,
        Banner,
        FullView,
        FloatView,
    }
}

File: iOS/TUIKit/TUICallKit/TUICallKit_Swift/State/ViewState.swift
//
//  ViewState.swift
//  Pods
//
//  Created by vincepzhang on 2025/2/6.
//

import RTCCommon

class ViewState: NSObject {
    
    let showLargeViewUserId: Observable<String> = Observable("")
    let isVirtualBackgroundOpened: Observable<Bool> = Observable(false)
    let router: Observable<ViewRouter> = Observable(.none)
    let callingViewType: Observable<CallingViewType> = Observable(.one2one)
    let isScreenCleaned: Observable<Bool> = Observable(false)
    
    enum ViewRouter {
        case none
        case banner
        case fullView
        case floatView
    }
    
    enum CallingViewType {
        case one2one
        case multi
    }
    
    func reset() {
        showLargeViewUserId.value = ""
        isVirtualBackgroundOpened.value = false
        router.value = .none
        callingViewType.value = .one2one
        isScreenCleaned.value = false
    }
}
